src/crypto/index.js

```diff
@@ -266,7 +270,10 @@ function _maybeUploadOneTimeKeys(crypto) {
         now - crypto._lastOneTimeKeyCheck < uploadPeriod
        ) {
         // we've done a key upload recently.
-        return;
+        if (currentKeyCount === undefined) {
+            // only return when currentCount (from /sync-response) is not set
```

Comment:
I changed to your suggested behavior. And I added a reset after all keys are uploaded. This should avoid this race.